Java/src/main/java/net/hypixel/api/util/PropertyFilter.java

```diff
@@ -0,0 +1,309 @@
+ * <p><br>
+ * To use an <strong>inclusion filter</strong>, property names (or "keys") can be added via {@link
+ * #include(String...) include(...)} or the {@link #with(String...) with(...) constructor}. When an
+ * object is {@link #applyTo(ComplexHypixelObject) passed through} the filter, any properties not
+ * explicitly named using the aforementioned methods will be removed from the object. If the object
+ * did not have an included property to begin with, it will not be created.
+ * <p><br>
+ * Property names are referenced using dot-notation. See the documentation for {@link
+ * #include(String...) include(...)} for more details.
+ */
+public class PropertyFilter {
+
+    /**
+     * Shorthand for constructing a new filter that only allows the {@code includedKeys} to {@link
+     * #applyTo(ComplexHypixelObject) pass through}. See {@link #include(String...)} for the key
+     * syntax.
+     */
+    public static PropertyFilter with(String... includedKeys) {
+        PropertyFilter filter = new PropertyFilter();
+        filter.include(includedKeys);
+        return filter;
+    }
+
+    // Only these keys are allowed in objects returned from `applyTo(...)`.
+    private final Set<PropertyKey> allowedKeys;
+
+    public PropertyFilter() {
+        allowedKeys = new HashSet<>();
+    }
+
+    /**
+     * Allows properties with any of the provided {@code keys} to {@link
+     * #applyTo(ComplexHypixelObject) pass through} the filter. To include nested properties, use
+     * dots ({@code .}) to separate each parent property from its child. If a property's name
+     * contains a dot literally, use a double-backslash to escape the dot. (e.g. {@code
+     * "key_with_literal_dot\\.in_it"} instead of {@code "key_with_literal_dot.in_it"})
+     * <br><pre>
+     * Examples:
+     *     •{@code uuid}                - Keep the player's UUID when filtering.
+     *     •{@code stats}               - Keep all of the player's stats when filtering.
+     *     •{@code stats.SkyWars}       - Keep all of the player's SkyWars stats when filtering.
+     *     •{@code stats.SkyWars.coins} - Keep just the player's SkyWars coins when filtering.
+     * </pre>
+     * Any duplicate keys, and keys already included in the filter, will not be added. If one key
+     * begins with the entirety of another key, and has more tokens / parts after that, the two keys
+     * are considered to be colliding. Some notes about key collision:
+     * <pre>
+     *     • New keys always take precedence when collision happens, such that...
+     *     • If an added key has a <u>wider scope</u> (fewer tokens) than any existing keys, all
+     *       colliding keys with narrower scopes (more tokens) will be removed, and the new key will
+     *       be added.
+     *     • If the added key has a <u>narrower scope</u> (more tokens) than an existing key, the
+     *       colliding key with a wider scope (fewer tokens) will be replaced with the new one.
+     * </pre>
```

Comment:
These are implementation details, really, our clients shouldn't care. It's self explanatory